Source/WebCore/rendering/updating/RenderTreeBuilderFormControls.cpp

```diff
@@ -99,4 +103,47 @@ RenderBlock& RenderTreeBuilder::FormControls::findOrCreateParentForChild(RenderM
     return *innerRenderer;
 }
 
+void RenderTreeBuilder::FormControls::updateCheckmark(RenderBlockFlow& flow)
+{
+    RefPtr inputElement = dynamicDowncast<HTMLInputElement>(flow.element());
+    ASSERT(inputElement);
+
+    auto pseudoStyle = flow.getCachedPseudoStyle({ PseudoElementType::Checkmark });
+    if (!pseudoStyle)
+        return;
+
+    for (CheckedRef child : childrenOfType<RenderElement>(flow)) {
+        if (child->style().pseudoElementType() == PseudoElementType::Checkmark) {
+            if (flow.style().appearance() != StyleAppearance::Base || pseudoStyle->display() == DisplayType::None)
+                m_builder.destroy(child);
+            return;
+        }
+    }
+
+    if (flow.style().appearance() != StyleAppearance::Base || pseudoStyle->display() == DisplayType::None)
+        return;
```

Comment:
`DisplayType::Contents` shouldn't create a box either. Are all other display property values ok?

Reply:
Actually `display:contents` gets changed to `none` in style adjuster so that comment can be ignored.